### ISSUE TITLE
Crash when returning from editing server details

### ISSUE THREAD

**Issue by twig** [erickok/transdroid#566]
Steps to reproduce:

* go to Settings
* click on your server info
* press back

Emulator image `Pixel 3a API 28`

```
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.transdroid.full, PID: 21531
    java.lang.RuntimeException: Unable to start activity ComponentInfo{org.transdroid.full/org.transdroid.core.gui.settings.MainSettingsActivity_}: androidx.fragment.app.Fragment$InstantiationException: Unable to instantiate fragment org.transdroid.core.gui.settings.PreferenceCompatActivity$RootPreferencesFragment: could not find Fragment constructor
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2913)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3048)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:78)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1808)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6669)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
     Caused by: androidx.fragment.app.Fragment$InstantiationException: Unable to instantiate fragment org.transdroid.core.gui.settings.PreferenceCompatActivity$RootPreferencesFragment: could not find Fragment constructor
        at androidx.fragment.app.Fragment.instantiate(Fragment.java:563)
        at androidx.fragment.app.FragmentContainer.instantiate(FragmentContainer.java:57)
        at androidx.fragment.app.FragmentManager$3.instantiate(FragmentManager.java:390)
        at androidx.fragment.app.FragmentStateManager.<init>(FragmentStateManager.java:74)
        at androidx.fragment.app.FragmentManager.restoreSaveState(FragmentManager.java:2454)
        at androidx.fragment.app.FragmentController.restoreSaveState(FragmentController.java:196)
        at androidx.fragment.app.FragmentActivity.onCreate(FragmentActivity.java:287)
        at androidx.appcompat.app.AppCompatActivity.onCreate(AppCompatActivity.java:106)
        at org.transdroid.core.gui.settings.MainSettingsActivity.onCreate(MainSettingsActivity.java:124)
        at org.transdroid.core.gui.settings.MainSettingsActivity_.onCreate(MainSettingsActivity_.java:40)
        at android.app.Activity.performCreate(Activity.java:7136)
        at android.app.Activity.performCreate(Activity.java:7127)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1271)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2893)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3048) 
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:78) 
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1808) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:193) 
        at android.app.ActivityThread.main(ActivityThread.java:6669) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858) 
     Caused by: java.lang.NoSuchMethodException: <init> []
        at java.lang.Class.getConstructor0(Class.java:2327)
        at java.lang.Class.getConstructor(Class.java:1725)
        at androidx.fragment.app.Fragment.instantiate(Fragment.java:548)
        at androidx.fragment.app.FragmentContainer.instantiate(FragmentContainer.java:57) 
        at androidx.fragment.app.FragmentManager$3.instantiate(FragmentManager.java:390) 
        at androidx.fragment.app.FragmentStateManager.<init>(FragmentStateManager.java:74) 
        at androidx.fragment.app.FragmentManager.restoreSaveState(FragmentManager.java:2454) 
        at androidx.fragment.app.FragmentController.restoreSaveState(FragmentController.java:196) 
        at androidx.fragment.app.FragmentActivity.onCreate(FragmentActivity.java:287) 
        at androidx.appcompat.app.AppCompatActivity.onCreate(AppCompatActivity.java:106) 
        at org.transdroid.core.gui.settings.MainSettingsActivity.onCreate(MainSettingsActivity.java:124) 
        at org.transdroid.core.gui.settings.MainSettingsActivity_.onCreate(MainSettingsActivity_.java:40) 
        at android.app.Activity.performCreate(Activity.java:7136) 
        at android.app.Activity.performCreate(Activity.java:7127) 
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1271) 
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2893) 
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3048) 
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:78) 
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1808) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:193) 
        at android.app.ActivityThread.main(ActivityThread.java:6669) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858) 
```

**Comment by erickok** [comment]
Thanks for the report.

**Comment by erickok** [comment]
This was fixed in df868dfa77c8826e5b34940733bad7d50e0a3ca2

**Comment by twig** [comment]
Thanks @erickok, seems to have fixed it
